cmd/podman/run.go

```diff
@@ -203,18 +202,15 @@ func runCmd(c *cli.Context) error {
 		return err
 	}
 
-	if ecode, err := ctr.ExitCode(); err != nil {
-		if errors.Cause(err) == libpod.ErrNoSuchCtr {
-			// The container may have been removed
-			// Go looking for an exit file
-			ctrExitCode, err := readExitFile(runtime.GetConfig().TmpDir, ctr.ID())
-			if err != nil {
-				logrus.Errorf("Cannot get exit code: %v", err)
-			} else {
-				exitCode = ctrExitCode
-			}
+	ecode, err := ctr.Wait()
+	if errors.Cause(err) == libpod.ErrNoSuchCtr {
```

Comment:
Stupid question because I'm unfamiliar with Go libraries/conventions: should this be wrapped inside a check for `err != nil`? Or does `errors.Cause()` guarantee proper handling of nil?